### ISSUE TITLE
Train/val split is not stratified — single card combinations can end up entirely in val

### ISSUE THREAD

**Issue by luipin** [luipin/set-ml-module#14]
### Summary
`SetCardDataModule.setup()` performs a plain random 80/20 split with no stratification. With only 81 unique card combinations, it is possible for some specific card combinations to land entirely in the validation set, meaning the model never trains on those combinations.

This is especially problematic when using the raw 81 seed images (not the bootstrapped augmented set), but it also affects bootstrapped datasets if augmented images from the same base card are not kept together.

### Code location
`src/data/set_card_data_pipeline.py`, line 149:

```python
train_paths, val_paths = train_test_split(all_images, test_size=0.2, random_state=42)
```

### Fix
Add a `stratify` parameter using the base card identity parsed from the filename:

```python
def _get_base_card(path):
    parts = os.path.basename(path).split('_')
    return '_'.join(parts[:4])  # color_shape_number_shading

card_ids = [_get_base_card(p) for p in all_images]
train_paths, val_paths = train_test_split(
    all_images, test_size=0.2, random_state=42, stratify=card_ids
)
```

This ensures each card combination is proportionally represented in both splits.

### Files affected
- `src/data/set_card_data_pipeline.py`
